ReactiveCocoa/Swift/Property.swift

```diff
@@ -90,16 +90,13 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	public var value: Value {
 		get {
 			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			defer { lock.unlock() }
+
+			return _value
 		}
 
 		set {
-			lock.lock()
-			_value = newValue
-			observer.sendNext(newValue)
-			lock.unlock()
+			modify { _ in newValue }
```

Comment:
But the setter won't return an old value that is read within the same critical section, and without it you will have to use `oldValue = modify { _ in newValue }` instead for the same purpose.